### ISSUE TITLE
Add of userfields in products

### ISSUE THREAD

**Issue by Will711990** [supergeorg/Grocy-SwiftUI#93]
Hi,

I added personal userfields in products, and in barcode:

<img width="780" alt="image" src="https://user-images.githubusercontent.com/25400986/152516464-7192c32a-d57c-4b6c-bad3-ffedb0c0146d.png">

Does it possible to integrate them in app ?

Thank you ! :)

**Comment by supergeorg** [comment]
It would be possible, but only via an implementation of custom user fields.

BTW: what do you want to do with this custom fields? Wouldn't it be easier if you realize this via unit conversions? They should work somehow (I haven't tested them throroughtly and you can only set them in the web gui for now) in the app and I am trying to improve its behavior.

**Comment by Will711990** [comment]
Hi, sorry for this really late answer.
I understand about implementation of custom user fields, maybe later :)

To answer to your question, I found `unit conversions ` not convenient in Grocy, because it depends on packaging of products.

For example:

Tomato sauce can be packaging:

- in bottle of 0,5L with a barcode on each bottle, or
- in a pack of 3 cartons of Tomato sauce with only 1 barcode for the 3 cartons

I would not create 2 products differents for these cases (pretty current).

 So I created Userfields:

- 1 userfield as list `Consumption`: for the product with choice 'Barcode' or 'Default Quantity'
- 1 userfield as case `Lot` (=Batch in english): For the barcode
- 1 userfield as Int Number `Nombre par lot` (=Number per batch in english): For the barcode

Also, if `Consumption = Barcode`, `batch is True`, and `Number per batch= 3` when I scan the barcode of batch it divides the quantity by Number of Batch.

Example: 1 barcode of 3 cartons of Tomato sauce of 0,120l each = total of 0,320L
When I scan the barcode as Consume, I remove only 0,120L, instead of 0,320L.

It's like a Unit conversion, but adapted for each barcode.

I adapted Barcode Buddy to realize this !

**Comment by supergeorg** [comment]
You can always add different purchase units and conversions to the product. This could be a "bottle" and a "pack" type, which converts to different values. 

But I am not sure if chained conversions are possible and supported (pack -> bottle -> ml).
The logic you described seems to be a bit hard to implement in the app (this needs scripting capabilities or the like), so your case with custom fields would not work anyways. But maybe a different solution is possible.

**Comment by supergeorg** [comment]
Yeah, it looks like a good thing, but this is still not so easy to implement. It needs custom dynamic parsing as well as custom UI for displaying it (and maybe even sorting it).
